session/session.go

```diff
@@ -1214,12 +1224,23 @@ func (s *session) IsCachedExecOk(ctx context.Context, prepared *ast.Prepared) (b
 		return false, nil
 	}
 	// maybe we'd better check cached plan type here, current
-	// only point select will be cached, see "getPhysicalPlan" func
-	return true, nil
+	// only point select/update will be cached, see "getPhysicalPlan" func
+	var ok bool
+	var err error
+	switch prepared.CachedPlan.(type) {
+	case *plannercore.PointGetPlan:
+		ok = true
+	case *plannercore.Update:
+		ok = true
+	default:
+		ok = false
+	}
+	return ok, err
 }
 
 // ExecutePreparedStmt executes a prepared statement.
 func (s *session) ExecutePreparedStmt(ctx context.Context, stmtID uint32, args []types.Datum) (sqlexec.RecordSet, error) {
+	s.PrepareTxnCtx(ctx)
```

Comment:
why change this?

Reply:
both cached execution and compile execution path need this, put here to make it simple